src/pypi_attestation_models/_impl.py

```diff
@@ -116,9 +116,12 @@ class AttestationPayload(BaseModel):
     @classmethod
     def from_dist(cls, dist: Path) -> AttestationPayload:
         """Create an `AttestationPayload` from a distribution file."""
+        with dist.open(mode="rb", buffering=0) as io:
+            digest = _sha256_streaming(io).hex()
```

Comment:
NB: This could be replaced with [`hashlib.file_digest`](https://docs.python.org/3/library/hashlib.html#hashlib.file_digest) in Python 3.11+.

Reply:
I added it as a comment